components/workflow-mgt/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/util/HumanTaskClientAPIAdminClient.java

```diff
@@ -0,0 +1,152 @@
+        try {
+            String headerString = WFImplConstant.MUTUAL_SSL_HEADER.replaceAll("\\$username", username);
+            mutualSSLHeader = AXIOMUtil.stringToOM(headerString);
+            serviceClient.addHeader(mutualSSLHeader);
+        } catch (XMLStreamException e) {
+            throw new AxisFault("Error while creating mutualSSLHeader XML Element.", e);
+        }
+        Options options = serviceClient.getOptions();
+        serviceClient.setOptions(options);
+    }
+
+    /**
+     * Lists the tasks matching the provided simple query object.
+     *
+     * @param queryInput : The simple query object with the filtering criteria.
+     * @return : The result set
+     * @throws java.rmi.RemoteException :
+     * @throws org.wso2.carbon.humantask.stub.ui.task.client.api.IllegalArgumentFault
+     *                                  :
+     * @throws org.wso2.carbon.humantask.stub.ui.task.client.api.IllegalStateFault
+     *                                  :
+     */
+    public TTaskSimpleQueryResultSet simpleQuery(TSimpleQueryInput queryInput)
+            throws RemoteException, IllegalArgumentFault, IllegalStateFault {
+        try {
+            return stub.simpleQuery(queryInput);
+        } catch (RemoteException | IllegalStateFault | IllegalArgumentFault e) {
+            log.error("Error occurred while performing taskListQuery operation", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Loads the task input.
+     *
+     * @param taskId : The id of the task/.
+     * @return : The task input OMElement.
+     * @throws RemoteException        :
+     * @throws IllegalStateFault      :
+     * @throws IllegalOperationFault:
+     * @throws IllegalAccessFault:
+     * @throws IllegalArgumentFault:
+     * @throws javax.xml.stream.XMLStreamException
+     * 
+     */
+    public OMElement getInput(URI taskId)
+            throws RemoteException, IllegalStateFault, IllegalOperationFault, IllegalAccessFault,
+            IllegalArgumentFault, XMLStreamException {
+        String errMsg = "Error occurred while performing loadTaskInput operation";
+        try {
+            String input = (String) stub.getInput(taskId, new NCName(""));
+            return AXIOMUtil.stringToOM(input);
+        } catch (RemoteException | IllegalStateFault | IllegalOperationFault | IllegalArgumentFault | IllegalAccessFault | XMLStreamException e) {
+            log.error(errMsg, e);
+            throw e;
+        }
+    }
+
+    /**
+     * The skip operation.
+     * @param taskId : The task id.
+     * @throws IllegalArgumentFault :
+     * @throws IllegalOperationFault  :
+     * @throws IllegalAccessFault :
+     * @throws IllegalStateFault :
+     * @throws RemoteException :
+     */
+    public void skip(URI taskId)
+            throws IllegalArgumentFault, IllegalOperationFault, IllegalAccessFault,
+            IllegalStateFault, RemoteException {
+        String errMsg = "Error occurred while performing skip operation";
```

Comment:
Any specific reason to defining the error message here instead of inside the catch block?